rfd/0038-database-access-aws-discovery.md

```diff
@@ -0,0 +1,163 @@
+---
+authors: Roman Tkachenko (roman@goteleport.com)
+state: draft
+---
+
+# RFD 38 - Automatic registration and discovery of databases on AWS
+
+## What
+
+Proposes a way for the database service agent to automatically discover and
+register AWS-hosted RDS, Aurora and Redshift databases.
+
+## Why
+
+Using database access with AWS-hosted databases is currently somewhat painful
+for a couple of reasons:
+
+- Users have to manually configure and attach IAM policies for each database
+  which is very error- and typo-prone, or create overly permissive IAM policies
+  which is insecure.
+- Users who have large numbers of AWS-hosted databases (we've had prospects with
+  hundreds) or create/delete them on demand have no good way to automatically
+  plug them into the cluster.
+
+Teleport database agent should be able to automatically do both of these.
+
+## Scope
+
+This RFD focuses on AWS-hosted databases. Similar ideas will be explored for
+GCP Cloud SQL in future.
+
+## IAM
+
+Currently, to allow IAM authentication with RDS instances, users have to create
```

Comment:
You also need to enable it on the instance. https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.Enabling.html

Reply:
I was under the impression that changing this setting requires instance restart which is why I didn't mention it but according to AWS docs, it looks like it can be applied immediately and without downtime which is cool. I'll include it in the RFD too then.

https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.DBInstance.Modifying.html#USER_ModifyInstance.Settings